src/components/home/balance-card.tsx

```diff
@@ -33,27 +33,43 @@ const BalanceCard = () => {
    * Fetch user's flextime data when it is undefined.
    */
   useEffect(() => {
-    if (!usersFlextime) {
+    // This card shows admin links if adminMode is true, no personal flextime for the admin themselves.
+    // So, only fetch if not adminMode and flextime data is not yet available.
+    if (!adminMode && !usersFlextime) {
       getUsersFlextimes();
+    } else if (adminMode) {
+      // If in admin mode, we are not displaying personal flextime, so ensure loading is false.
+      setLoading(false);
     }
```

Comment:
This seems weird and I think it won't be needed if we fix the issue with setting the loading state below.